aptos-move/move-examples/tic-tac-toe/sources/TicTacToe.move

```diff
@@ -0,0 +1,639 @@
+module tic_tac_toe::ttt {
+    use std::vector;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::error;
+    use aptos_framework::account;
+    use aptos_framework::event;
+
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+
+
+
+    //// GAME CONSTANTS
+    const GRID_SIZE: u64 = 9;
+    const WIDTH_AND_HEIGHT: u64 = 3;
+    const EMPTY_CELL: u64 = 3;
+
+    //// PLAYER CONSTANTS
+    const PLAYER_X_TYPE: u64 = 1;
+    const PLAYER_O_TYPE: u64 = 2;
+
+    //// ERROR CODES
+    const ERR_ILLEGAL_ACTION: u64 = 0;
+    const ERR_INVALID_MOVE: u64 = 1;
+    const ERR_PLAYER_TAKEN: u64 = 2;
+    const ERR_GAME_NOT_DONE: u64 = 3;
```

Comment:
FYI, the standard for error codes are pretty simple, and add useful messages to errors in the explorer and elsewhere such as:

```
/// Error message that will show when it fails in the transaction
ENAME_OF_ERROR: u64 = 1;
```